app/com/yetu/oauth2provider/services/data/MemoryAuthCodeAccessTokens.scala

```diff
@@ -12,117 +14,56 @@ import scalaoauth2.provider.AccessToken
+    logger.debug(s"saveAuthCode code=$code authInfo=$authInfo")
+    Future.successful(authCodes += (code -> authInfo))
   }
 
-  def findAccessTokenByUser(user: YetuUser): Option[AccessToken] = {
-
-    logger.debug(s"findAccessTokenByUser user=$user")
-    accessTokensWithUser.foreach {
-      case (key, value) => {
-        logger.debug(value.user.identityId.userId + " - " + user.identityId)
-        if (value.user.identityId.userId == user.identityId.userId) {
-          return Some(key)
-        }
-      }
-    }
-    None
+  def saveAccessTokenToAuthInfo(token: String, authInfo: AuthInfo[YetuUser]) = {
+    logger.debug(s"saveAuthCode token=$token authInfo=$authInfo")
+    Future.successful(accessTokensToAuthInfo += (token -> authInfo))
   }
 
-  def findTokenByAccessToken(accessToken: AccessToken): Option[String] = {
-    logger.debug(s"findTokenByAccessToken accessToken=$accessToken")
-    accessTokens.foreach {
-      case (key, value) => {
-        if (value.equals(accessToken)) {
-          return Some(key)
-        }
-      }
-    }
-    None
+  def saveAuthInfoToAccessToken(key: String, accessToken: AccessToken): Future[Unit] = {
+    logger.debug(s"saveAuthInfoToAccessToken key: $key, accessToken: $accessToken")
+    Future.successful(accessTokens += (key -> accessToken))
   }
 
-  def deleteAll(identity: YetuUser) {
-    try {
-      val authCode = findAuthCodeByUser(identity)
-      authCodes --= authCode
-    } catch {
-      case e: Exception => logger.debug("exception caught: " + e);
-    }
-
-    try {
-      val accessToken = findAccessTokenByUser(identity)
-      accessToken match {
-        case None => None
-        case Some(a) => {
-          val token = findTokenByAccessToken(a)
-          accessTokens --= token
-        }
-      }
-      accessTokensWithUser --= accessToken
-    } catch {
-      case e: Exception => logger.debug("exception caught: " + e);
-    }
-  }
-
-  def saveAuthCodeToAuthInfo(code: String, authInfo: AuthInformation) = {
-    logger.debug(s"saveAuthCodeToAuthInfo code: $code, authInfo: $authInfo")
-    authCodeAuthInfo += (code -> authInfo)
-  }
-
-  def findAuthInfoByAuthCode(code: String): Option[AuthInformation] = {
-    val info = authCodeAuthInfo.get(code)
+  def findAuthInfoByAuthCode(code: String) = {
+    val info = authCodes.get(code)
     logger.debug(s"findAuthInfoByAuthCode code: $code result=$info")
-    info
+    Future.successful(info)
   }
 
-  def saveAccessTokenToUser(accessToken: AccessToken, authInfo: AuthInformation) = {
-    logger.debug(s"saveAccessTokenToUser accessToken: $accessToken")
-    logger.debug(s"saveAccessTokenToUser authInfo: $authInfo")
-    accessTokensWithUser += (accessToken -> authInfo)
+  def findAuthInfoByAccessToken(token: String) = {
+    val authInfo: Option[AuthInfo[YetuUser]] = accessTokensToAuthInfo.get(token)
+    logger.debug(s"findUserByAccessToken token: $token")
+    logger.debug(s"findUserByAccessToken authInfo: $authInfo")
```

Comment:
Previously you logged `token` and `authInfo` in one single line. You might want to use the same pattern.